### PR TITLE
Migrate to null safety

### PR DESCRIPTION
**Pull request by natebosch**

### CODE REVIEW

File: lib/src/digest_sink.dart [hunk 1]
```diff
@@ -7,24 +7,20 @@ import 'digest.dart';
...
   @override
   void add(Digest value) {
-    assert(_value == null);
     _value = value;
   }
 
   @override
   void close() {
-    assert(_value != null);
+    assert((_value as dynamic) != null);
```
Comment (jakemac53) [inline lib/src/digest_sink.dart:24]: probably worth adding a comment about this being a workaround for an assignment check on a late final field

File: pubspec.yaml [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
+version: 2.2.0-nullsafety
 description: Implementations of SHA, MD5, and HMAC cryptographic functions
 homepage: https://www.github.com/dart-lang/crypto
 
 environment:
-  sdk: '>=2.3.0 <3.0.0'
+  sdk: '>=2.10.0-2.0.dev <2.10.0'
 
 dependencies:
   collection: '^1.0.0'
```
Comment (jakemac53) [inline pubspec.yaml:10]:
```suggestion
  collection: ^1.15.0-nullsafety
```

File: pubspec.yaml [hunk 3]
```diff
@@ -14,3 +14,61 @@ dependencies:
 dev_dependencies:
   pedantic: ^1.0.0
   test: ^1.0.0
```
Comment (jakemac53) [inline pubspec.yaml:16]:
```suggestion
  pedantic: ^1.10.0-nullsafety
  test: ^1.16.0-nullsafety
```

File: pubspec.yaml [hunk 4]
```diff
@@ -14,3 +14,61 @@ dependencies:
...
+  test_core:
+    git:
+      url: git://github.com/dart-lang/test.git
+      path: pkgs/test_core
+  test:
+    git:
+      url: git://github.com/dart-lang/test.git
+      path: pkgs/test
+  typed_data:
+    git: git://github.com/dart-lang/typed_data.git
```
Comment (jakemac53) [inline pubspec.yaml:74]:
```suggestion
```

File: lib/src/digest_sink.dart (hunk 1 above)
Comment (natebosch) [reply lib/src/digest_sink.dart:25]: Sure.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,14 @@
+## 2.2.0-nullsafety
+
+Pre-release for the null safety migration of this package.
+
+Note that 2.2.0 may not be the final stable null safety release version, we
```
Comment (kevmoo) [inline CHANGELOG.md:5]: I'm SO GLAD we're being clear here. Thanks!

CC @jakemac53

Comment (natebosch) [reply CHANGELOG.md:5]: I just copy/pasted this from Jake's changelogs. Thanks Jake!

**Pull request merged**